drivers/storage/scaleio/tests/scaleio_test.go

```diff
@@ -320,9 +334,9 @@ func TestVolumeAttach(t *testing.T) {
 		vol = volumeCreate(t, client, volumeName)
 		_ = volumeAttach(t, client, vol.ID)
 		_ = volumeInspectAttached(t, client, vol.ID)
-		_ = volumeInspectDetachedFail(t, client, vol.ID)
+		//_ = volumeInspectDetachedFail(t, client, vol.ID) TODO: has the behavior changed?
```

Comment:
Yes, it definitely did, but it's been a couple months. :)  The notion of when attachments should be returned very much depends on the query, and whether you've asked for attachments, attachmentsToMe, attachmentsWithDevices, etc.

here's an example of where I fixed this in EBS a couple weeks ago: https://github.com/codedellemc/libstorage/pull/500/files#diff-5812461cc888fb891da3e63c5d1433f0R214